SIPS/sip-12.md

```diff
@@ -51,12 +51,14 @@ This permission is specified as follows in `snap.manifest.json` files:
   "initialPermissions": {
     "endowment:name-lookup": {
         "chains": ["eip155:1", "bip122:000000000019d6689c085ae165831e93"],
+        "tlds": [".lens"],
     }
   }
 }
 ```
 
-`chains` - An array of CAIP-2 chain IDs that the snap supports. This field is useful for a client in order to avoid unnecessary overhead.
+`chains` - A non-empty array of CAIP-2 chain IDs that the snap supports. This field is useful for a client in order to avoid unnecessary overhead.
+`tlds` - A non-empty array of top level domains that the snap will provide resolution for. This field is useful for a client for validating input for domain resolution, also helpful in reducing overhead.
```

Comment:
I'm wondering if this will turn out problematic if we wanna support resolution for formats that don't use TLDs? Should we make this optional?

Reply:
@FrederikBolding I agree this should be optional. Also see my comment here: https://github.com/MetaMask/SIPs/discussions/103#discussioncomment-7995271 as it relates to supporting naming system which use character standards different from ENS.

Reply:
Yes, that is the intention. Please see my comment @pawmmm: https://github.com/MetaMask/SIPs/discussions/103#discussioncomment-8015612

Reply:
@hmalik88 If it is intended to be optional, let's make that clear in the SIP.
```suggestion
`tlds` - An optional non-empty array of top level domains that the snap will provide resolution for. This field is useful for a client for validating input for domain resolution, also helpful in reducing overhead.
```

Reply:
Ah yeah I've been meaning to push an update -- will get to it later today, I think I might change the name of this property and mention that it needs to be a non-empty array of either tlds or non-tlds and then include some language to define some prefix standard for indicating a non-tld lookup.